### DISCUSSION TITLE
Specify arm64 at the start

### DISCUSSION THREAD

**Discussion by jasonrandrews** [aws/copilot-cli#5492] [Q&A]
Is there a way to specify arm64 and deploy with 1 command? 

I would like to add the platform to this command:

`copilot init --app demo 
  --name api 
  --type 'Load Balanced Web Service' 
  --dockerfile './Dockerfile' 
  --deploy`

**Comment by Lou1415926** [comment]
hi @jasonrandrews ! `copilot init` under the hoods executes `copilot svc init`, which will figure out your machine's os/arch, and if it's different than the default, it'll put them into the manifest that it generates. Then `copilot init` will execute `copilot svc deploy`, which will consume the manifest being generated. The expectation is that your service will be deployed with the same os/arch config as your local machine. Was the behavior that you observed different than the expected behavior?

Alternatively - if your `api` always uses the same manifest, you can add the os/arch configuration into the manifest, and then just run `copilot init` from this point - `copilot svc init` will not attempt to override your existing manifest, and `copilot svc deploy` will respect the os/arch config from your manifest.

**Reply by jasonrandrews** [reply]
Hi @Lou1415926, thanks for the reply. I will check it again, but I ran copilot init on a Linux machine with aarch64 arch, and it did not set the platform to linux/arm64. I wrote up the instructions [Use AWS Graviton processors on AWS Fargate with Copilot](https://dev.to/aws-builders/use-aws-graviton-processors-on-aws-fargate-with-copilot-1n6p) and would appreciate any corrections or feedback.

**Reply by Lou1415926** [reply]
My bad! `copilot svc init` indeed explicitly write `linux/x86_64` once it detects that your os/arch is _not_ `linux/amd64`.  You are correct that updating the manifest's `platform` field is necessary in order to use an arm-based image for the service.
